pkg/runtime/pubsub/cloudevents.go

```diff
@@ -0,0 +1,30 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package pubsub
+
+import (
+	contrib_pubsub "github.com/dapr/components-contrib/pubsub"
+	"github.com/google/uuid"
+)
+
+// CloudEvent is a reqeust object to create a Dapr compliant cloudevent
+type CloudEvent struct {
+	ID              string
+	Data            []byte
+	Topic           string
+	Pubsub          string
+	DataContentType string
+	TraceID         string
+}
+
+// NewCloudEvent encapusalates the creation of a Dapr cloudevent from an existing cloudevent or a raw payload
+func NewCloudEvent(req *CloudEvent) (map[string]interface{}, error) {
+	if req.DataContentType == contrib_pubsub.ContentType {
+		return contrib_pubsub.FromCloudEvent(req.Data, req.Topic, req.Pubsub, req.TraceID)
```

Comment:
Can you add a test where this line is covered? It seems to be critical.

Reply:
Yep

Reply:
Done